### PR TITLE
Fix grpc server

### PR DESCRIPTION
**Pull request by joshklop**
Use grpc.Serve, not http.ListenAndServe.

### CODE REVIEW

**Comment by joshklop** [comment]
Hmm, appears the grpc server hangs on shutdown. Will look into it.

**Review by kirugan — APPROVED** [review]

**Comment by kirugan** [comment]
> Hmm, appears the grpc server hangs on shutdown. Will look into it.

Josh the reason was: even if you stop grpc server it returns nil from `Serve(...)` and then goroutines block on write to `errCh`. I added bufferization and now it works.

P.S. Still not sure about `ErrServerStopped` though - in grpc source code it's said that `// Serve called after Stop or GracefulStop.` for this error.

**Comment by joshklop** [comment]
Thanks! I replaced the check for `http.ErrServerStopped` with `err != nil` and made the channel unbuffered. Mind taking another look real quick?

File: node/http.go [hunk 1]
```diff
@@ -81,10 +82,46 @@ func makeMetrics(port uint16) *httpService {
...
 }
 
-func makeGRPC(port uint16, database db.DB, version string) *httpService {
-	grpcHandler := grpc.NewServer()
-	gen.RegisterKVServer(grpcHandler, junogrpc.New(database, version))
-	return makeHTTPService(port, grpcHandler)
+type grpcService struct {
+	srv  *grpc.Server
+	port uint16
+}
```
Comment (omerfirmak) [inline node/http.go:88]: this probably belongs to `grpc` package

**Pull request merged**